pkg/client/request.go

```diff
@@ -87,3 +92,17 @@ func marshal(v interface{}) (*http.Request, error) {
 
 	return r, nil
 }
+
+func ContextWithReason(ctx context.Context, reason fmt.Stringer) context.Context {
+	return context.WithValue(ctx, reasonContextKey, reason.String())
+}
+
+func ReasonFromContext(ctx context.Context) (string, error) {
+	var reason string
+	if r := ctx.Value(reasonContextKey); r != nil {
+		if rr, ok := r.(string); ok {
+			reason = rr
+		}
```

Comment:
Shouldn't we return an error saying the `CruiseControlRequestReason` key not found from context?

Reply:
Nice catch. Pushed the wrong version. Fixed.